openeo/extra/spectral_indices/spectral_indices.py

```diff
@@ -0,0 +1,99 @@
+from openeo.processes import ProcessBuilder, array_modify
+from openeo.rest.datacube import DataCube
+import numpy as np
+
+def _get_expression_map(cube: DataCube, x: ProcessBuilder):
+    collection_id = cube.graph["loadcollection1"]["arguments"]["id"]
+    bands = [band.replace("0", "").upper() for band in cube.metadata.band_names]
+
+    def check_validity():
+        if not all(band in orig_bands for band in bands):
+            raise ValueError(
+                "The bands in your cube {} are not all Sentinel-2 bands (the following are: {})".format(bands,orig_bands))
+
+    def get_params():
+        lookup = dict(zip(orig_bands, norm_bands))
+        return {lookup[key]: x.array_element(i) for i, key in enumerate(bands)}
+
+    def lookupS2():
+        orig_bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7", "B8", "B8A", "B9", "B11", "B12"]
+        norm_bands = ["A", "B", "G", "R", "RE1", "RE2", "RE3", "N", "RE4", "WV", "S1", "S2"]
+        return orig_bands, norm_bands
+
+    def lookupL8():
+        orig_bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7", "B10", "B11"]
+        norm_bands = ["A", "B", "G", "R", "N", "S1", "S2", "T1", "T2"]
+        return orig_bands, norm_bands
+
+    def lookupL457():
+        orig_bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7"]
+        norm_bands = ["B", "G", "R", "N", "S1", "T1", "S2"]
+        return orig_bands, norm_bands
+
+    def lookupModis():
+        orig_bands = ["B3", "B4", "B1", "B2", "B5", "B6", "B7"]
+        norm_bands = ["B", "G", "R", "N", np.nan, "S1", "S2"]
+        return orig_bands, norm_bands
+
+    def lookupProbav():
+        orig_bands = ["BLUE", "RED", "NIR", "SWIR"]
+        norm_bands = ["B", "R", "N", "S1"]
+        return orig_bands, norm_bands
+
+    lookupPlatform = {
+        "LANDSAT4-5_TM_L1": lookupL457,
+        "LANDSAT4-5_TM_L2": lookupL457,
+        "LANDSAT8_L1": lookupL8,
+        "LANDSAT8_L2": lookupL8,
+        "MODIS": lookupModis,
+        "PROBAV_L3_S5_TOC_100M": lookupProbav,
+        "PROBAV_L3_S10_TOC_333M": lookupProbav,
+        "SENTINEL2_L1C_SENTINELHUB": lookupS2,
+        "SENTINEL2_L2A_SENTINELHUB": lookupS2,
+        "TERRASCOPE_S2_TOC_V2": lookupS2,
+    }
+
+    if collection_id not in list(lookupPlatform.keys()):
+        raise Exception("Sorry, satellite platform not supported for index computation!")
+
+    orig_bands, norm_bands = lookupPlatform[collection_id]()
```

Comment:
this whole setup looks bit overly complicated:
- define `lookupX()` functions that return static list
-  put `lookupX()` functions in dict
- get desired `lookupX()` from dict
- call `lookupX()` to get     `orig_bands, norm_bands`
- use `dict(zip(orig_bands, norm_bands))`  to convert to mapping


I think you can just define a dictionary that holds the orig-to-common band name mappings